debezium-schema-generator/src/main/java/io/debezium/schemagenerator/maven/SchemaGeneratorMojo.java

```diff
@@ -48,74 +27,24 @@
 
     @Override
-    public void execute() throws MojoExecutionException, MojoFailureException {
-        String classPath = getClassPath();
-
-        try {
-            int result = exec(SchemaGenerator.class.getName(), classPath, Collections.emptyList(), Arrays.<String> asList(format, outputDirectory.getAbsolutePath()));
-
-            if (result != 0) {
-                throw new MojoExecutionException("Couldn't generate API spec; please see the logs for more details");
-            }
-            getLog().info("Generated API spec at " + outputDirectory.getAbsolutePath());
-        }
-        catch (IOException | InterruptedException e) {
-            throw new MojoExecutionException("Couldn't generate API spec", e);
-        }
-    }
-
-    private int exec(String className, String classPath, List<String> jvmArgs, List<String> args) throws IOException, InterruptedException {
-        String javaHome = System.getProperty("java.home");
-        String javaBin = javaHome + File.separator + "bin" + File.separator + "java";
-
-        List<String> command = new ArrayList<>();
-        command.add(javaBin);
-        command.addAll(jvmArgs);
-        command.add("-cp");
-        command.add(classPath);
-        command.add(className);
-        command.addAll(args);
-
-        ProcessBuilder builder = new ProcessBuilder(command);
-        Process process = builder.inheritIO().start();
-        process.waitFor();
-
-        return process.exitValue();
-    }
-
-    private String getClassPath() {
-        Set<Artifact> artifacts = project.getDependencyArtifacts();
-
-        String classPath = artifacts.stream()
-                .filter(a -> a.getScope().equals(Artifact.SCOPE_COMPILE) || a.getScope().equals(Artifact.SCOPE_PROVIDED))
-                .map(a -> a.getFile().getAbsolutePath())
-                .collect(Collectors.joining(File.pathSeparator));
-
-        classPath += classPathEntryFor(getClass());
-        classPath += classPathEntryFor(OpenAPI.class);
-        classPath += classPathEntryFor(ConfigDef.class);
-        classPath += classPathEntryFor(Format.class);
-        classPath += classPathEntryFor(DebeziumException.class);
-        classPath += classPathEntryFor(JsonProcessingException.class);
-        classPath += classPathEntryFor(YAMLFactory.class);
-        classPath += classPathEntryFor(JsonNode.class);
-        classPath += classPathEntryFor(JsonView.class);
-        classPath += classPathEntryFor(DotName.class);
-        classPath += classPathEntryFor(Charsets.class);
-
-        classPath += File.pathSeparator + project.getArtifact().getFile().getAbsolutePath();
-
-        return classPath;
-    }
-
-    private String classPathEntryFor(Class<?> clazz) {
-        return File.pathSeparator + clazz.getProtectionDomain().getCodeSource().getLocation().toString();
+    public void execute() {
+        new SchemaGenerator().run(format, outputDirectory.toPath(), groupDirectoryPerConnector, filenamePrefix, filenameSuffix);
```

Comment:
@gunnarmorling I changed the Maven Plugin to work inlined instead of that separate JVM process.